gui/src/logger.rs

```diff
@@ -117,4 +125,39 @@ impl Logger {
         self.level_handle.modify(|filter| *filter = log_level)?;
         Ok(())
     }
+
+    pub fn receiver(&self) -> crossbeam_channel::Receiver<String> {
+        self.receiver.clone()
+    }
+}
+
+/// Used as a layer to send log messages to a channel.
+pub struct LogStream {
+    pub sender: crossbeam_channel::Sender<String>,
+}
+
+impl<S> layer::Layer<S> for LogStream
+where
+    S: Subscriber + for<'a> tracing_subscriber::registry::LookupSpan<'a>,
+{
+    fn on_event(&self, event: &Event<'_>, _ctx: layer::Context<'_, S>) {
+        let mut visitor = LogStreamVisitor {
+            sender: &self.sender,
+        };
+        event.record(&mut visitor);
+    }
+}
+
+/// Used to record log messages by sending them to the channel.
+struct LogStreamVisitor<'a> {
+    sender: &'a crossbeam_channel::Sender<String>,
+}
+
+impl<'a> tracing::field::Visit for LogStreamVisitor<'a> {
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn std::fmt::Debug) {
+        if field.name() == "message" {
+            let msg = format!("{:?}", value);
+            let _ = self.sender.send(msg);
```

Comment:
We should check error indicating that the other side of the channel is disconnected in order to stop sending message in it.

Reply:
I haven't found a way to use the error to tell this layer to stop being triggered. I was trying to set a flag of some kind, but the `on_event` method only takes `&self` so I didn't manage to.

Reply:
With my latest changes, we may not need to do anything here as we only send messages to the channel while `start_bitcoind_and_daemon` is running.